pkg/controller/utils/component.go

```diff
@@ -192,6 +193,52 @@ func (c componentHandler) CreateOrUpdate(ctx context.Context, component render.C
 	return nil
 }
 
+func (c componentHandler) Delete(ctx context.Context, component render.Component, status status.StatusManager) error {
+	// Before deleting the component, make sure that it is ready. This provides a hook to do
+	// dependency checking for the component.
+	cmpLog := c.log.WithValues("component", reflect.TypeOf(component))
+	cmpLog.V(2).Info("Checking if component is ready")
+	if !component.Ready() {
```

Comment:
I think there might be some nuance here that I'm not understanding about this function vs. the `CreateOrUpdate` function above (besides the fact that CreateOrUpdate should probably be called CreateOrUpdateOrDelete)

Reply:
Delete returns an error if the component is not installed. I added a check to see if the component is marked as ready (installation was successful), then I can delete objects that reside within the tigera-*<component>* namespace.

Reply:
Does not apply anymore.